morf-core/src/main/java/org/alfasoftware/morf/sql/element/FieldLiteral.java

```diff
@@ -42,24 +44,126 @@
+   */
+  public static FieldLiteral literal(final String stringValue) {
+    return new FieldLiteral("", stringValue, DataType.STRING);
+  }
+
+
+  /**
+   * @param value A stringified value.
+   * @param dataType The data type represented by the stringified value.
+   * @return An appropriately typed literal.
+   */
+  public static FieldLiteral literal(final String value, final DataType dataType) {
+    return new FieldLiteral("", value, dataType);
+  }
+
 
-    this.value = sourceField.value;
-    this.dataType = sourceField.dataType;
+  /**
+   * @param doubleValue Any double.
+   * @return A numeric literal.
+   */
+  public static FieldLiteral literal(final Double doubleValue) {
+    return new FieldLiteral("", doubleValue != null ? doubleValue.toString() : null, DataType.DECIMAL);
+  }
+
+
+  /**
+   * @param bigDecimalValue Any decimal.
+   * @return A numeric literal.
+   */
+  public static FieldLiteral literal(final BigDecimal bigDecimalValue) {
+    return new FieldLiteral("", bigDecimalValue != null ? bigDecimalValue.toPlainString() : null, DataType.DECIMAL);
+  }
+
+
+  /**
+   * @param integerValue Any integer.
+   * @return An integer literal.
+   */
+  public static FieldLiteral literal(final Integer integerValue) {
+    return new FieldLiteral("", integerValue != null ? integerValue.toString() : null, DataType.DECIMAL);
+  }
+
+
+  /**
+   * @param localDateValue Any date.
+   * @return A date literal.
+   */
+  public static FieldLiteral literal(final LocalDate localDateValue) {
+    return new FieldLiteral("", localDateValue != null ? localDateValue.toString("yyyy-MM-dd") : null, DataType.DATE);
+  }
+
+
+  /**
+   * @param longValue Any long.
+   * @return A big integer literal.
+   */
+  public static FieldLiteral literal(final Long longValue) {
+    return new FieldLiteral("", longValue != null ? longValue.toString() : null, DataType.DECIMAL);
+  }
+
+
+  /**
+   * @param charValue Any character.
+   * @return A string literal.
+   */
+  public static FieldLiteral literal(final Character charValue) {
+    return new FieldLiteral("", charValue != null ? charValue.toString() : null, DataType.STRING);
+  }
+
+
+  /**
+   * @param booleanValue Any boolean..
+   * @return A boolean literal.
+   */
+  public static FieldLiteral literal(final Boolean booleanValue) {
+    return new FieldLiteral("", booleanValue != null ? booleanValue.toString() : null, DataType.BOOLEAN);
+  }
+
+
+  /**
+   * TODO make private when {@link NullFieldLiteral} is removed
+   *
+   * @param alias The alias.
+   * @param value The value.
+   * @param dataType The data type.
+   */
+  protected FieldLiteral(final String alias, final String value, final DataType dataType) {
+    super(alias);
+    this.value = value;
+    this.dataType = dataType;
   }
 
 
   /**
    * Constructs a new {@linkplain FieldLiteral} with as string source.
    *
    * @param stringValue the literal value to use
+   * @deprecated Use {@link #literal(String)}

```

Comment:
Shouldn't this refer to the Utils class? (There's a lot of these)

Reply:
Depends whether we think these static methods belong here and whether we actually want to deprecate use of the constructors.  Are they harmful?  Might make life easier to not deprecate them, then there's no value in adding these static methods. Thoughts?